code/_onclick/hud/fullscreen.dm

```diff
@@ -0,0 +1,107 @@
+#define FULLSCREEN_LAYER 18
+#define DAMAGE_LAYER FULLSCREEN_LAYER + 0.1
+#define BLIND_LAYER DAMAGE_LAYER + 0.1
+#define CRIT_LAYER BLIND_LAYER + 0.1
+
+/mob
+	var/list/screens = list()
+
+/mob/proc/overlay_fullscreen(category, type, severity)
+	var/obj/screen/fullscreen/screen
+	if(screens[category])
+		screen = screens[category]
+		if(screen.type != type)
+			clear_fullscreen(category, FALSE)
+			return .()
+		else if(!severity || severity == screen.severity)
+			return null
+	else
+		screen = PoolOrNew(type)
+
+	screen.icon_state = "[initial(screen.icon_state)][severity]"
+	screen.severity = severity
+
+	screens[category] = screen
+	if(client)
+		client.screen += screen
+	return screen
+
+/mob/proc/clear_fullscreen(category, animate = 10)
+	var/obj/screen/fullscreen/screen = screens[category]
+	if(!screen)
+		return
+
+	if(animate)
+		animate(screen, alpha = 0, time = animate)
+		sleep(animate)

```

Comment:
Are you sure this is safe? this can lock up controllers etc if called carelessly. it might make more sense to spawn() something off here instead.